test/refactoring/ConvertAsync/convert_async_attributed_renames.swift

```diff
@@ -0,0 +1,95 @@
+// REQUIRES: concurrency
+
+// RUN: %empty-directory(%t)
+
+@available(*, renamed: "simple2")
+func simple(_ completion: @escaping (String) -> Void) { }
+@available(*, renamed: "simple2")
+func nonCompletionName(_ random: @escaping (String) -> Void) { }
+func simple2() async -> String { }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=SIMPLERENAMED %s
+func simpleRenamed() {
+  // preserve me
+  simple { str in
+    print(str)
+  }
+  // and me
+  nonCompletionName { str in
+    print(str)
+  }
+}
+// SIMPLERENAMED: func simpleRenamed() async {
+// SIMPLERENAMED-NEXT: // preserve me
+// SIMPLERENAMED-NEXT: let str = await simple2()
+// SIMPLERENAMED-NEXT: print(str)
+// SIMPLERENAMED-NEXT: // and me
+// SIMPLERENAMED-NEXT: let str1 = await simple2()
+// SIMPLERENAMED-NEXT: print(str1)
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=RENAMEDCOLLISION %s
+func renamedCollision() {
+  simple { simple2 in
+    print(simple2)
+  }
+}
+// RENAMEDCOLLISION: func renamedCollision() async {
+// RENAMEDCOLLISION-NEXT: let simple21 = await simple2()
+// RENAMEDCOLLISION-NEXT: print(simple21)
+
+@available(*, renamed: "simpleArg2")
+func simpleArgRenamed(arg: String, _ random: @escaping (String) -> Void) { }
+@available(*, renamed: "simpleArg2")
+func completionFirstArg(random: @escaping (String) -> Void, arg: String) { }
+func simpleArg2(newArg: String) async -> String { }
+
+// RUN: %refactor-check-compiles -convert-to-async -dump-text -source-filename %s -pos=%(line+1):6 | %FileCheck -check-prefix=ARGRENAMED %s
+func argRenamed() {
+  (simpleArgRenamed)(arg: "foo") { str in
+    print(str)
+  }
+  (completionFirstArg)(random: { str in
+    print(str)
+  }, arg: "foo")
+}
+// ARGRENAMED: func argRenamed() async {
+// ARGRENAMED-NEXT: let str = await (simpleArg2)(newArg: "foo")
+// ARGRENAMED-NEXT: print(str)
+// ARGRENAMED-NEXT: let str1 = await (simpleArg2)(newArg: "foo")
+// ARGRENAMED-NEXT: print(str1)
+
+@available(*, renamed: "multiHandlers2")
+func multiHandlers(arg: String, handler1: @escaping (String) -> Void, handler2: @escaping (String) -> Void) { }
```

Comment:
Would it also make sense to add a test case that has two completion-like parameters and a parameter afterwards?
```swift
@available(*, renamed: "multiHandlersWithTrailing2")
func multiHandlersWithTrailing(arg: String, handler1: @escaping (String) -> Void, handler2: @escaping (String) -> Void, other: String) {}
func multiHandlers2(newArg: String, newHandler: @escaping (String) -> Void, other: String) async -> String { }
```

Reply:
More can't hurt!